diff --git a/cmake/protobuf.cmake b/cmake/protobuf.cmake
index 438dff720c5..47863c17e23 100644
--- a/cmake/protobuf.cmake
+++ b/cmake/protobuf.cmake
@@ -205,19 +205,22 @@ MACRO(MYSQL_CHECK_PROTOBUF)
   # Version 22 and up depend on ~65 abseil .dylibs.
   IF(APPLE AND WITH_PROTOBUF STREQUAL "system" AND
       PB_MINOR_VERSION VERSION_GREATER 21)
     # list(FILTER <list> {INCLUDE | EXCLUDE} REGEX <regex>)
     FIND_OBJECT_DEPENDENCIES("${PROTOBUF_LIBRARY}" protobuf_dependencies)
-    LIST(FILTER protobuf_dependencies INCLUDE REGEX "${HOMEBREW_HOME}.*")
+    LIST(FILTER protobuf_dependencies INCLUDE REGEX "${HOMEBREW_HOME}.*|${CMAKE_INSTALL_PREFIX}.*")
+    LIST(FILTER protobuf_dependencies EXCLUDE REGEX "^/System/Library/|^/usr/lib/")
     SET_TARGET_PROPERTIES(ext::libprotobuf PROPERTIES
       INTERFACE_LINK_LIBRARIES "${protobuf_dependencies}"
       )
     FIND_OBJECT_DEPENDENCIES("${PROTOBUF_LITE_LIBRARY}" lite_dependencies)
-    LIST(FILTER lite_dependencies  INCLUDE REGEX "${HOMEBREW_HOME}.*")
+    LIST(FILTER lite_dependencies  INCLUDE REGEX "${HOMEBREW_HOME}.*|${CMAKE_INSTALL_PREFIX}.*")
+    LIST(FILTER lite_dependencies  EXCLUDE REGEX "^/System/Library/|^/usr/lib/")
     SET_TARGET_PROPERTIES(ext::libprotobuf-lite PROPERTIES
       INTERFACE_LINK_LIBRARIES "${lite_dependencies}"
       )
     FIND_OBJECT_DEPENDENCIES("${Protobuf_PROTOC_LIBRARY}" protoc_dependencies)
-    LIST(FILTER protoc_dependencies INCLUDE REGEX "${HOMEBREW_HOME}.*")
+    LIST(FILTER protoc_dependencies INCLUDE REGEX "${HOMEBREW_HOME}.*|${CMAKE_INSTALL_PREFIX}.*")
+    LIST(FILTER protoc_dependencies EXCLUDE REGEX "^/System/Library/|^/usr/lib/")
     SET_TARGET_PROPERTIES(ext::libprotoc PROPERTIES
       INTERFACE_LINK_LIBRARIES "${protoc_dependencies}"
       )
